ydb/core/scheme/scheme_type_info.cpp

```diff
@@ -11,14 +12,41 @@ ::TString TypeName(const TTypeInfo typeInfo, const ::TString& typeMod) {
     case NScheme::NTypeIds::Pg:
         return NPg::PgTypeNameFromTypeDesc(typeInfo.GetPgTypeDesc(), typeMod);
     case NScheme::NTypeIds::Decimal: {
-        const TDecimalType decimal = typeInfo.GetDecimalType();
-        if (decimal.GetPrecision() == DECIMAL_PRECISION && decimal.GetScale() == DECIMAL_SCALE)
-            return "Decimal";
+        const TDecimalType& decimal = typeInfo.GetDecimalType();
         return Sprintf("Decimal(%u,%u)", decimal.GetPrecision(), decimal.GetScale());
     }
     default:
         return TypeName(typeInfo.GetTypeId());
     }
 }
 
+bool GetTypeInfo(const NScheme::IType* type, const NKikimrProto::TTypeInfo& typeInfoProto, const TStringBuf& typeName, const TStringBuf& columnName, NScheme::TTypeInfo &typeInfo, ::TString& errorStr) {
+    if (type) {
+        // Only allow YQL types
+        if (!NScheme::NTypeIds::IsYqlType(type->GetTypeId())) {
+            errorStr = Sprintf("Type '%s' specified for column '%s' is no longer supported", typeName.data(), columnName.data());
+            return false;
+        }
+        typeInfo = NScheme::TTypeInfo(type->GetTypeId());
+        return true;
+    } else if (const auto decimalType = NScheme::TDecimalType::ParseTypeName(typeName)) {
+        if (typeInfoProto.HasDecimalPrecision() && typeInfoProto.HasDecimalScale()) {
+            typeInfo = NScheme::TypeInfoFromProto(NScheme::NTypeIds::Decimal, typeInfoProto);
+        } else {
+            typeInfo = *decimalType;
```

Comment:
What does it mean? Why isn't it error case? Can you explain for me please?

Reply:
There are some normal cases where proto is empty.

Reply:
Can you give me some examples please? Or even put some explanation comment in the source code?

Reply:
Let's close as a duplicate of https://github.com/ydb-platform/ydb/pull/9587/files#r1779368945